sinonjs/samsam
# Remove use of .callee

This PR removes use of `.callee` in order to get closer to 100% test coverage.

While looking at the coverage report, I noticed that we were mocking `engineCanDetectArguments ` with `proxyquire` in this test, even when the source no longer uses it.

When I set out to tidy it up, I discovered that IE11 can detect `arguments`.

Keep reading ...

#### Purpose

Improve test coverage

![2019-10-10 at 10 22](https://user-images.githubusercontent.com/20321/66556334-ed6c1680-eb47-11e9-8612-b81af8cbfbfb.png)

#### Background

IE11 is the only IE supported in **all** Sinon projects.

In IE11, calling toString on an arguments object returns "[object Arguments]", just like all the modern engines. I verified this by modifying a test and running it in IE11 on SauceLabs.

Therefore, it is safe to remove the branch that uses `.callee` for detecting the arguments object.

This helps us get closer to 100% test coverage.

#### Solution  - optional

* Remove the branching in `isArguments`
* Simplify tests for `isArguments`

![2019-10-10 at 10 23](https://user-images.githubusercontent.com/20321/66556414-17bdd400-eb48-11e9-8a98-dcc05d0ceefa.png)

#### New Major release

Because drops support for very old IE, it has potential to break non-Sinon dependents. We should release this commit in a new major version.

#### How to verify - mandatory

1. Check out this branch
1. `npm install`
1. Add a `.only` to the `describe` in `is-arguments.test.js` file, so only those tests are run
1. `npm run test-cloud`
1. Observe that tests pass in all supported engines.
1. Observe that there are no skipped tests

```
# internet explorer 11:


  ......

  6 passing (7ms)
```

#### Checklist for author
